### PR TITLE
ENH: prevent 'from __future__ import annotations'

### LINKED ISSUE #241: Report from __future__ import annotations
As far as I understand, `from __future__ import annotations` is not needed in stub files. It would be great to report this unnecessary import statement.

### PR DESCRIPTION
**Pull request by twoertwein**
closes #241

### CODE REVIEW

File: tests/imports.pyi [hunk 1]
```diff
@@ -3,6 +3,9 @@
 # Note: DO NOT RUN ISORT ON THIS FILE.
 # It's excluded in our pyproject.toml.
 
+# BAD IMPORTS (Y044)
+from __future__ import annotations  # Y044 "from __future__ import annotations" has no effect in stub files.
```
Comment (twoertwein) [inline tests/imports.pyi:7]: Can also move it to the bad imports. I put it up there as I think `__future__` imports have to be the very first import.

Comment (AlexWaygood) [reply tests/imports.pyi:7]: Having it here is fine I think.

File: README.md [hunk 2]
```diff
@@ -74,6 +74,7 @@ currently emitted:
 | Y040 | Never explicitly inherit from `object`, as all classes implicitly inherit from `object` in Python 3. This error code is incompatible with stubs supporting Python 2.
 | Y041 | Y041 detects redundant numeric unions. For example, PEP 484 specifies that type checkers should treat `int` as an implicit subtype of `float`, so `int` is redundant in the union `int \| float`. In the same way, `int` is redundant in the union `int \| complex`, and `float` is redundant in the union `float \| complex`.
 | Y043 | Do not use names ending in "T" for private type aliases. (The "T" suffix implies that an object is a `TypeVar`.)
+| Y044 | "from __future__ import annotations" has no effect in stub files.
```
Comment (AlexWaygood) [inline README.md:77]:
```suggestion
| Y044 | `from __future__ import annotations` has no effect in stub files, as forward references in stubs are enabled by default.
```

**Review by AlexWaygood — APPROVED** [review]
Thanks!

**Pull request merged**